### PR TITLE
swap -  bugz pass #1

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes TEAM2-38 TEAM2-37 TEAM2-7 TEAM2-6

## What changed (plus any additional context for devs)
I fixed all the things

- few missed spots for l2 token icons + colors(related to swaps)
- fixed some exchange row icon, fallback + formatting
- rewrote part of the disabling output for arbitrum logic so we still use the input mask
- mixed minimum sig figs for minimum received so its human readable

## PoW (screenshots / screen recordings)
https://cloud.skylarbarrera.com/Screen-Recording-2022-06-08-12-41-14.mp4

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/components/exchange/ExchangeField.js [hunk 1]
```diff
@@ -85,47 +88,45 @@ const ExchangeField = (
 
   return (
     <Container {...props}>
-      <TouchableWithoutFeedback onPress={handleFocusField}>
+      <TouchableWithoutFeedback
+        onPress={onTapWhileDisabled || handleFocusField}
```
Comment (estebanmino) [inline src/components/exchange/ExchangeField.js:92]:
```suggestion
        onPress={onTapWhileDisabled ?? handleFocusField}
```

File: src/hooks/useSwapAdjustedAmounts.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export default function useSwapAdjustedAmounts(tradeDetails: Quote) {
   const address = inputAsExact
     ? outputCurrency.mainnet_address || outputCurrency.address
     : inputCurrency.mainnet_address || inputCurrency.address;
-  const priceValue = genericAssets[address]?.price?.value ?? 0;
+  const priceValue = genericAssets[address]?.price?.value ?? 0.0000000001;
```
Comment (estebanmino) [inline src/hooks/useSwapAdjustedAmounts.ts:42]: wondering what's this number coming from?

File: src/components/exchange/ExchangeField.js [hunk 3]
```diff
@@ -85,47 +88,45 @@ const ExchangeField = (
...
+            testID={testID}
+            useCustomAndroidMask={useCustomAndroidMask}
+            value={amount}
+          />
         </FieldRow>
       </TouchableWithoutFeedback>
       {!disableCurrencySelection && (
         <TokenSelectionButton
           address={address}
+          mainnet_address={mainnet_address}
```
Comment (brunobar79) [inline src/components/exchange/ExchangeField.js:125]: you're going to hell for writing a prop like this

File: src/components/exchange/ExchangeInputField.js [hunk 4]
```diff
@@ -44,6 +45,7 @@ export default function ExchangeInputField({
         amount={inputAmount}
         disableCurrencySelection={disableInputCurrencySelection}
         editable={editable}
+        mainnet_address={inputCurrencyMainnetAddress}
```
Comment (brunobar79) [inline src/components/exchange/ExchangeInputField.js:48]: to hell back and forth twice

File: src/components/exchange/ExchangeInputField.js [hunk 5]
```diff
@@ -68,6 +70,7 @@ export default function ExchangeInputField({
         <ExchangeMaxButton
           address={inputCurrencyAddress}
           disabled={!inputCurrencySymbol}
+          mainnet_address={inputCurrencyMainnetAddress}
```
Comment (brunobar79) [inline src/components/exchange/ExchangeInputField.js:73]: 👿

File: src/hooks/useSwapAdjustedAmounts.ts (hunk 2 above)
Comment (brunobar79) [reply src/hooks/useSwapAdjustedAmounts.ts:42]: me too

**Review by brunobar79 — APPROVED** [review]
straight_2_hell

File: src/components/exchange/ExchangeField.js (hunk 3 above)
Comment (skylarbarrera) [reply src/components/exchange/ExchangeField.js:125]: in my defense u started it

File: src/hooks/useSwapAdjustedAmounts.ts (hunk 2 above)
Comment (skylarbarrera) [reply src/hooks/useSwapAdjustedAmounts.ts:42]: lmao look away I meant to change this

Comment (skylarbarrera) [reply src/hooks/useSwapAdjustedAmounts.ts]: was a placeholder for token price, replaced with the correct way to get: `ethereumUtils.getAssetPrice`

**Review by estebanmino — APPROVED** [review]

**Pull request merged**